numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -0,0 +1,364 @@
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#define _MULTIARRAYMODULE
+
+#include "npy_pycompat.h"
+#include "get_attr_string.h"
+#include "npy_import.h"
+#include "multiarraymodule.h"
+
+
+/*
+ * Get an object's __array_function__ method in the fastest way possible.
```

Comment:
Can you extend this comment to explain what a `NULL` means (ie, will an exception never / sometimes / always be set)